### ISSUE TITLE
Connected SG not checking in to sensorgnome.org

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#634]
I have two RPi SG's right now configured to connect to my home network.

One of them checked in immediately to sensorgnome.org and has been syncing data. (Though it appears as an old terminated deployment name.)

The other never checked in or showed up on sensorgnome.org. Though when I just checked, it is connected to my WAN and I can access both SG's directly by IP address over the WAN.

They are both running the same software version, both using the onboard WiFi adapter, and both connected to to WAN, yet only one is checking in.

**Comment by jsayers-bsc** [comment]
SG that is connecting and syncing properly: SG-91EBRPI3E583

![image](https://user-images.githubusercontent.com/55001144/69495776-a1c3c100-0e98-11ea-9253-eab978295046.png)

SG that is connected to my WAN but not checking in: SG-0FBARPI359DA

![image](https://user-images.githubusercontent.com/55001144/69495786-bef88f80-0e98-11ea-9a9b-278244c132af.png)

**Comment by adamdsmith** [comment]
I’ve experienced the same issue. In fact, when I disconnect the SG that first registered and connect any other, it connects properly but shows up on the network as the originally-connected SG. But when you click through on sensorgnome.org, it takes you to the appropriate web interface of the new SG. So every time I confirm connectivity for a new SG, I have to navigate through the serial number of the originally-registered SG. 

Upon re-read,  that sounds a bit convoluted. Let me know if you something needs clarified.

**Comment by jsayers-bsc** [comment]
That is strange indeed! 

Though I think something else is going on here. I had actually connected the problematic SG first, then the other one. But upon reading your comment I tried again to connect them in that (same) order, though the result was the same.  

I also wondered whether other issues might be contributing. There was no GPS location originally on the SG that wouldn't check in, despite being powered on for nearly 20 hours. I rebooted and the GPS hat readily acquired a location this time around. But no difference there; it was still available on my wireless network, but didn't check in.

I also noticed that that SG didn't have the "Live pulses" and "what I'm doing now" status visible (which you can see in the previous screenshot). That's something I've noticed here and there, and it seems to be somewhat hit or miss. But in this case it was due to their being no dongles attached. Attaching those resolved that issue, but not the original one.

Then I just swapped the SD cards between the two, in case there had been some inadvertent modification to the software made at some point. No difference there either: it was still the same physical receiver that connected despite the software swap.

And I tried plugging in to Ethernet directly to see if that made a difference, but it didn't. (Admittedly I did not modify the network.txt after plugging into Ethernet, though I don't _think_ that would matter.)

Given all that, it would appear to be more related to the physical device than anything else, though that doesn't make sense either considering it is still connecting to my WiFi. 

While that all remains a mystery, I guess I know which one is getting deployed tomorrow!

**Comment by adamdsmith** [comment]
Yeah that’s odd. I do notice these odd little blips from time to time with the SG software that a reboot often solves. It’s usually not a big deal but there is the occasional loss of a couple months of data. 

BTW, I missed that you were connecting via WiFi initially. My issue was with SGs connected simultaneously via Ethernet to the same router. Maybe that helps explain why only one SG could get out on my end.  I wonder then if the SensorGnome server then assigns the serial number of the first SG connecting to a given IP? I don’t recall that always being the case.

**Comment by leberrigan** [comment]
Does the SG even try to sync with the SG server if there aren't any dongles attached? Maybe try connecting a dongle for a day and see if the status changes over that time.

**Comment by jsayers-bsc** [comment]
@leberrigan From what I can tell it does, though I'm not positive...

I didn't think this was a software related issue because if I swapped the SD cards between the two units, the same one would connect, and the same one wouldn't.

However when I got out in the field to deploy the functional one (still with the other SD card, which didn't seem to have any ill effect at my home), it would not connect. When I swapped back in its original SD card, it did connect.

So when I got home I wiped the SD clean and copied over a fresh unadulterated copy of the software and tried again. This time the SG that had never connected did connect. 

So it may actually be something obliquely related to what @adamdsmith mentioned: " I wonder then if the SensorGnome server then assigns the serial number of the first SG connecting to a given IP?" I'm not sure what was going on here, but it seems that the connection to the original RPi worked, even with the other "bad" SD card, because it was still connecting on the same network that it had connected to originally. When it tried to connect to a _different_ network with that same "bad" SD card, it failed. 

Though that still doesn't explain why this RPi didn't connect with the "good" SD card. I guess it's still something of a mystery!

But the moral of the story in this case was: try again with a completely fresh software version and go from there.
